### DISCUSSION TITLE
Activity not mandatory

### DISCUSSION THREAD

**Discussion by niels-przybilla** [kimai/kimai#4721] [General]
Hello Kimai Community,
I am testing out Kimai. Is it possible that Activity is not mandatory? Usually I wam working only with tasks and descriptions.
The activity is from my current workflow a nice to have but not mandatory.

many thanks fpr clarification.

Thank you and all the best

Niels

**Comment by kevinpapst** [comment]
No. But you can create one global activity, which will then be auto-selected.
Using the Custom-CSS plugin, you can hide the field (see documentation).

**Reply by pkoevesdi** [reply]
I want to join this request. 
We have several Activities, but none of them is supposed to be mandatory for our users. They should be allowed to leave "Activities" empty. 

> you can create one global activity, which will then be auto-selected.

I made one, but it isn't auto-selected. How can I achieve this?

Actually, same feature request for projects. There are works, which are not mapped to a project. I think, in every company You have project-independent work packages, like book-keeping. But of course, You still want to track the book-keepers work too.

**Reply by pkoevesdi** [reply]
Is it possible anyway to have more than one activity, and still make one of them default?. That would be a workaround for not being able to make these not mandatory.
I thought, some Javascript could to, but these field are dynamically generated, so it's not easy, is it?

**Reply by kevinpapst** [reply]
There are several javascript examples: https://www.kimai.org/documentation/plugin-custom-content.html

**Reply by pkoevesdi** [reply]
Ah, ok, nice, didn't find them, this works. Thank You!

**Comment by niels-przybilla** [comment]
It would be beautiful to have this as a setting to have activity optional or mandatory :-) THX

**Reply by pkoevesdi** [reply]
How can this be adopted for the "Create for other user" form as well? (In German GUI: "erstellen" when on the "Alle Zeiten" page.)

**Comment by EduNetEurope** [comment]
It would be great if we could select for clients and or projects if activities are mandatory.

**Reply by kMikaZu** [reply]
Would definitely be handy...
